src/webview_provider/queryResultPanel.ts

```diff
@@ -306,6 +313,25 @@ export class QueryResultPanel extends AltimateWebviewProvider {
     return project;
   }
 
+  private async executeQueryFromActiveWindow(message: { limit: number }) {
+    const activeEditor = window.activeTextEditor;
+    if (!activeEditor) {
+      window.showErrorMessage("No active editor found");
+      return;
+    }
+    const query = activeEditor.document.getText();
```

Comment:
I think this should take into account selected text similar to the existing implementation.